shared/middleware/files_api.rb

```diff
@@ -117,6 +140,28 @@ def allowed_file_type?(endpoint, extension)
   end
 
   #
+  # PUT /v3/(assets|sources)/<channel-id>?abuse_score=<abuse_score>
```

Comment:
in REST language, `/v3/(assets|sources)/<channel-id>` is a collection and a PUT to the collection should replace the entire collection. Because you are updating the entire collection based on the query param, I think the design you've chosen is nice and RESTful.

Nit: according to https://en.wikipedia.org/wiki/Representational_state_transfer, there should be a trailing slash (after `<channel-id>`) when updating a collection.